### PR TITLE
Spec Updates

### PR DESCRIPTION
**Pull request by nebhale**
This change updates the library to match upcoming spec changes.  It adds the Platform to Detect, updates the Detect logger to use stdout, updates Detect's build plan output to write to \<plan>, and updates Build to allow claims on build plan dependencies via \<plan>.

### CODE REVIEW

File: command/riff_invoker.go [hunk 1]
```diff
@@ -115,9 +117,16 @@ func NewCommandInvoker(build libjavabuildpack.Build) (RiffCommandInvoker, bool,
...
-		launch:        build.Launch,
-		invokerLayer:  build.Launch.DependencyLayer(dep),
-		functionLayer: build.Launch.Layer("function"),
+		layers:        build.Layers,
+		invokerLayer:  build.Layers.DependencyLayer(dep),
+		functionLayer: build.Layers.Layer("function"),
 	}, true, nil
+}
+
+type marker struct {
```
Comment (ericbottard) [inline command/riff_invoker.go]: I think this also needs to contain the language of the function, in the weird case where you would 
- change the language of the function
- keep the same artifact
- the value of the FUNCTION_URI should not be the same
(very very unlikely I admit, but...)

Comment (nebhale) [reply command/riff_invoker.go]: Yep, totally legit.

**Comment by nebhale** [comment]
This is ready for merge and will work with `pack` 0.0.8 and latest images.

**Comment by ericbottard** [comment]
Closing in favor of https://github.com/projectriff/riff-buildpack/pull/41.

Thanks @nebhale !

**Pull request closed without merging**